app/services/generate_test_data.rb

```diff
@@ -60,4 +62,13 @@ def fake_provider
       code: 'ABC',
     )
   end
+
+  RETRY_COUNT_TIMES = 20
+
+  def random_code(klass, provider)
+    RETRY_COUNT_TIMES.times do
```

Comment:
maybe with could use a loop without need to use a constant RETRY_COUNT_TIMES:
a `while` or `repeat` with an exit condition.

Reply:
I've put `RETRY_COUNT_TIMES` in there because there is a limit to the number of unique codes you can have so we could exhaust the possible combinations in which case you'd end up in an infinite loop.